### PR TITLE
added working ul endpoint

### PR DESCRIPTION
**Pull request by isemona**

### CODE REVIEW

File: apps/api/src/main.ts [hunk 1]
```diff
@@ -22,7 +31,8 @@ app.use(session({
   cookie: {
     http: false,
     sameSite: 'lax'
-  }
+  },
+  store
```
Comment (alisaduncan) [inline apps/api/src/main.ts:35]: Are you planning on building onto the session store? If not, why not instantiate here in this file?

File: apps/api/src/universalLogout.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  // 204 When the request is successful
+  const httpStatus = 204;
+
+  // 400 If the request is malformed
+  if (!req.body) {
+    res.status(400);
+  }
+
+  // Find the user by email linked to the org id associated with the API key provided
+  const domainOrgId = req['user']['id']
```
Comment (alisaduncan) [inline apps/api/src/universalLogout.ts]:
```suggestion
  const domainOrgId = req['user']['id'];
```

File: apps/api/src/universalLogout.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  const userId = user.id;
+  const sids = [];
+  Object.keys(storedSession).forEach((key) => {
+    const sess = JSON.parse(storedSession[key]);
+    if (sess.passport.user === userId) {
+      sids.push(key);
+    }
+  });
+
+  sids.map((sid) => store.destroy(sid));
```
Comment (alisaduncan) [inline apps/api/src/universalLogout.ts]: i highly recommend (again) that you don't use map here. Do for...of

map implies you're mapping the sid to something else, it's a functional method. But you're not doing that in this operation.

File: apps/todo-app/src/app/components/todolist.tsx [hunk 4]
```diff
@@ -29,6 +29,15 @@ export const Todos = () => {
           body: JSON.stringify({ task: newTask })
         });
 
+        if (!res.ok)
+          {if (res.status === 401) {
+            // Redirect user back to the sign-in page
+            window.location.href = '/';
```
Comment (alisaduncan) [inline apps/todo-app/src/app/components/todolist.tsx:35]: recommend using React instead of Window API?

**Review by alisaduncan — COMMENTED** [review]
Some comments

**Review by alisaduncan — APPROVED** [review]

File: apps/api/src/main.ts (hunk 1 above)
Comment (isemona) [reply apps/api/src/main.ts:35]: discussed to keep as is

File: apps/todo-app/src/app/components/todolist.tsx (hunk 4 above)
Comment (isemona) [reply apps/todo-app/src/app/components/todolist.tsx:35]: suggested using useNavigate

**Pull request merged**